source/slang/ir-insts.h

```diff
@@ -112,63 +163,69 @@ struct IRInterpolationModeDecoration : IRDecoration
 /// names, emit debug information, etc.
 struct IRNameHintDecoration : IRDecoration
 {
-    enum { kDecorationOp = kIRDecorationOp_NameHint };
+    enum { kOp = kIROp_NameHintDecoration };
+    IR_LEAF_ISA(NameHintDecoration)
 
-    Name* name;
+    IRStringLit* getNameOperand() { return cast<IRStringLit>(getOperand(0)); }
+
+    UnownedStringSlice getName()
```

Comment:
Don't know if we want to assert if getNameOperand result is not null.

Reply:
The `cast` operation that `getNameOperand` is using here is a checked cast that will assert if the object isn't of the correct type, so there should be no way for it to return null. The naming isn't very good, and we may want to have a pass where we systematically rename things with something like `dynCast`, `checkedCast`, etc. to express the intent better.